src/Data/Either/Plucky.hs

```diff
@@ -459,49 +460,49 @@ catch e k = case e of
 -- necessary.
 --
 -- @since 0.0.0.0
-rethrow :: e -> Either e a
-rethrow = Left
+rethrow :: MonadError e m => e -> m a
+rethrow = throwError
 
 -- | Like 'catch', but promoted to the 'ExceptT' monad transformer.
 --
 -- @since 0.0.0.0
 catchT
-    :: (Monad m)
+    :: (MonadError e' m)
     => ExceptT e m a
-    -> (e -> ExceptT e' m a)
-    -> ExceptT e' m a
-catchT action handler = ExceptT $ do
+    -> (e -> m a)
+    -> m a
+catchT action handler = do
     ea <- runExceptT action
     case ea of
         Left err ->
-            runExceptT $ handler err
+            handler err
         Right a ->
-            pure (Right a)
+            pure a
 
 
 -- | Like 'catchOne', but promoted to the 'ExceptT' monad transformer.
 --
 -- @since 0.0.0.1
 catchOneT
-    :: (Monad m)
+    :: (MonadError e' m)
     => ExceptT (Either e e') m a
-    -> (e -> ExceptT e' m a)
-    -> ExceptT e' m a
-catchOneT action handler = ExceptT $ do
+    -> (e -> m a)
+    -> m a
+catchOneT action handler = do
```

Comment:
Yeah, this works!

My only problem with it is that I have a `CheckedT` transformer in [`prio`](https://github.com/parsonsmatt/prio) that takes advantage of this technique, and I'd probably prefer to use it to "pluck" the type out since it has better instances wrt `MonadIO` etc. But I'd also want users to be able to write it with `ExceptT`  as a concrete type, too, for obvious ecosystem reasons.

I need to play with this more. I've got the next week on vacation and maybe I'll spend some time working on this then!